purchase_order_product_recommendation/wizards/purchase_order_recommendation.py

```diff
@@ -0,0 +1,290 @@
+            )
+            self.line_ids += new_line
+            # Limit number of results. It has to be done here, as we need to
+            # populate all results first, for being able to select best matches
+            i += 1
+            if i == self.line_amount:
+                break
+
+    @api.multi
+    def action_accept(self):
+        """Propagate recommendations to purchase order."""
+        po_lines = self.env['purchase.order.line']
+        sequence = max(self.order_id.mapped('order_line.sequence') or [0])
+        for wiz_line in self.line_ids.filtered('is_modified'):
+            # Use preexisting line if any
+            if wiz_line.purchase_line_id:
+                if wiz_line.units_included:
+                    wiz_line.purchase_line_id.update({
+                        'product_qty': wiz_line.units_included,
+                    })
+                    wiz_line.purchase_line_id._onchange_quantity()
+                else:
+                    wiz_line.purchase_line_id.unlink()
+                continue
+            sequence += 1
+            # Use a new in-memory line otherwise
+            po_line = po_lines.new({
+                'order_id': self.order_id.id,
+                'product_id': wiz_line.product_id.id,
+                'sequence': sequence,
+            })
+            po_line.onchange_product_id()
+            po_line.product_qty = wiz_line.units_included
+            po_line._onchange_quantity()
+            po_lines |= po_line
+        self.order_id.order_line |= po_lines
+
+
+class PurchaseOrderRecommendationLine(models.TransientModel):
+    _name = 'purchase.order.recommendation.line'
+    _description = 'Recommended product for current purchase order'
+    _order = 'id'
+
+    currency_id = fields.Many2one(
+        related='product_id.currency_id',
+        readonly=True,
+    )
+    partner_id = fields.Many2one(
+        related='wizard_id.order_id.partner_id',
+        readonly=True,
+    )
+    product_id = fields.Many2one(
+        'product.product',
+        string='Product',
+    )
+    price_unit = fields.Monetary(
+        compute='_compute_price_unit',
+    )
+    times_delivered = fields.Integer(
+        readonly=True,
+    )
+    times_received = fields.Integer(
+        readonly=True,
+    )
+    units_received = fields.Float(
+        readonly=True,
+    )
+    units_delivered = fields.Float(
+        readonly=True,
+    )
+    units_max_delivered = fields.Float(
+        readonly=True,
+    )
+    units_avg_delivered = fields.Float(
+        digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True,
+    )
+    units_available = fields.Float(
+        readonly=True,
+    )
+    units_virtual_available = fields.Float(
+        readonly=True,
+    )
+    units_included = fields.Float()
+    wizard_id = fields.Many2one(
+        'purchase.order.recommendation',
+        'Wizard',
+        ondelete='cascade',
+        required=True,
```

Comment:
same here

Reply:
same as https://github.com/OCA/purchase-workflow/pull/660#discussion_r260253622

Reply:
Also, same as here ??

Reply:
Sure, a wizard line without it's parent model would be useless :wink: